lib/presentation/widgets/icon_list.dart

```diff
@@ -1,32 +1,27 @@
 import 'package:flutter/material.dart';
-import 'package:trip_tally/presentation/theme/app_dimensions.dart';
-import 'package:trip_tally/presentation/widgets/custom_icon_button.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:trip_tally/presentation/utils/enums/transport_methods.dart';
 
 class IconList extends StatelessWidget {
-  const IconList({required this.icons, super.key});
+  const IconList({
+    required this.icons,
+    required this.onIconSelected,
+    super.key,
+  });
 
   final List<String> icons;
+  final ValueChanged<TransportMethods> onIconSelected;
```

Comment:
Good to know about `ValueChanged`. I never saw this one and it's a typedef for a generic Function(TransportMethods method)